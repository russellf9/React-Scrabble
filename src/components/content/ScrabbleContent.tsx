import * as React from 'react';
import styled from 'styled-components';
import { colors, fonts, fontWeights } from '../../themes';

export interface ContentProps  {
    search: string;
}

interface ContentState {
    search: string;
}

export const Title = styled.h2`
    color: ${colors.darkBlue};
    font-family: ${fonts.hind};
    font-size: 1.75em;
    font-weight: ${fontWeights.semiBold};
    text-align: center;
    text-transform: uppercase;
`;

const Wrapper = styled.section`
    background: ${colors.background};
    font-family: 'Hind', sans-serif;
    padding: 4em;
`;

export class ScrabbleContent extends React.Component<ContentProps, ContentState> {
    constructor(props: ContentProps) {
        super(props);
    }

    public render() {
        return (
            <Wrapper>
                <Title>Scrabble</Title>
            </Wrapper>
        );
    }
}