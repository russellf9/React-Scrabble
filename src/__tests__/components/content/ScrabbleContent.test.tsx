import * as React from 'react';
import { shallow } from 'enzyme';
import { ContentProps, ScrabbleContent } from '../../../components/content/ScrabbleContent';

describe('ScrabbleContent', () => {

  const props: ContentProps = {
    search: '',
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<ScrabbleContent {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });  
});
