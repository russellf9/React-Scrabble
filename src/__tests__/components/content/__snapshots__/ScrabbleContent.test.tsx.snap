// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrabbleContent renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrabbleContent
    search=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED>
        Scrabble
      </UNDEFINED>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Scrabble",
      },
      "ref": null,
      "rendered": "Scrabble",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    color: ",
            "#2993C8",
            ";
    font-family: ",
            "hind",
            ";
    font-size: 1.75em;
    font-weight: ",
            "600",
            ";
    text-align: center;
    text-transform: uppercase;
",
          ],
        },
        "displayName": "styled.h2",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "h2",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
    background: ",
          "#f8f9fa",
          ";
    font-family: 'Hind', sans-serif;
    padding: 4em;
",
        ],
      },
      "displayName": "styled.section",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "section",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED>
          Scrabble
        </UNDEFINED>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Scrabble",
        },
        "ref": null,
        "rendered": "Scrabble",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
    color: ",
              "#2993C8",
              ";
    font-family: ",
              "hind",
              ";
    font-size: 1.75em;
    font-weight: ",
              "600",
              ";
    text-align: center;
    text-transform: uppercase;
",
            ],
          },
          "displayName": "styled.h2",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "h2",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
    background: ",
            "#f8f9fa",
            ";
    font-family: 'Hind', sans-serif;
    padding: 4em;
",
          ],
        },
        "displayName": "styled.section",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
